File: backend/src/main/java/com/beadforge/model/entity/OfficialMessage.java
package com.beadforge.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("t_official_message")
public class OfficialMessage {
    @TableId(type = IdType.AUTO)
    private Long id;

    /** OFFICIAL / ACTIVITY */
    private String channel;

    private String title;
    private String content;
    private String icon;
    private String color;

    private LocalDateTime publishedAt;

    private Integer enabled;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/officialMessage.ts
import client from './client';

interface ApiRes<T> {
  code: number;
  message: string;
  data: T;
}

export interface OfficialMessageItem {
  id: string;
  rawId: number;
  channel: 'OFFICIAL' | 'ACTIVITY';
  title: string;
  content: string;
  icon: string;
  color: string;
  time: string;
}

export const officialMessageApi = {
  list: (channel?: 'OFFICIAL' | 'ACTIVITY') =>
    client.get<any, ApiRes<OfficialMessageItem[]>>('/official-messages', {
      params: channel ? { channel } : undefined,
    }),
};
